pkg/agent/run.go

```diff
@@ -58,9 +57,9 @@ func setupCriCtlConfig(cfg cmds.Agent, nodeConfig *daemonconfig.Node) error {
 		}
 	}
 
-	agentConfDir := datadir.DefaultDataDir + "/agent/etc"
+	agentConfDir := filepath.Join(cfg.DataDir, "agent", "etc")
 	if _, err := os.Stat(agentConfDir); os.IsNotExist(err) {
-		if err := os.MkdirAll(agentConfDir, 0755); err != nil {
+		if err := os.MkdirAll(agentConfDir, 0700); err != nil {
```

Comment:
/etc/ is typically 0755 which this emulates. granted that the parent is 0700 this change doesn't amount to much

Reply:
I've tested this and it doesn't seem to be a problem, and is now consistent with the other agent dir that we create below.